AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java

```diff
@@ -312,14 +312,21 @@ private void updateParticipantViews() {
         }
 
         gridLayout.post(() -> {
-            if ((prevParticipantViewList.size() > 1 && participantViewList.size() <= 1)
-                    || (prevParticipantViewList.size() <= 1 && participantViewList.size() > 1)) {
+            final int preSize = prevParticipantViewList.size();
+            final int currSize = participantViewList.size();
+            if (!rangeInDefined(preSize, 0, 1) && rangeInDefined(currSize, 0, 1)
+                || (!rangeInDefined(preSize, 2, 4) && rangeInDefined(currSize, 2, 4))
+                || (!rangeInDefined(preSize, 5, 6)) && rangeInDefined(currSize, 5, 6)) {
```

Comment:
My initial feeling is that this could be improved to be more understandable, but it's not a blocker.

Reply:
Yea, probably it can be improved. but I have not find a better one to replace it. Basically we need a simpler way to quick look up the current number in which range then map it to the corresponding grid layout.